src/app/main/main.component.ts

```diff
@@ -290,6 +287,24 @@ export class MainComponent implements OnInit, OnDestroy {
     return dryRun;
   }
 
+  // 'count' is used to track the number of request. Each request is 10MB, so we are limiting the total to 250 MB.
+  getResults(count: number, projectId: string, inputPageToken: string, location: string, jobId: string)  : Promise<BigQueryResponse> {
+    if (!inputPageToken || count >= 25) {
+      // Force an update feature here since everything is done.
+      var localRows : Array<Object> = [];
+      localRows.push(...this.rows);
+      this.rows = localRows;
+      return;
+    }
+    count = count + 1;
+    return this.dataService.getResults(projectId, jobId, location, inputPageToken, this.columns, this.stats).then(({ rows, stats, pageToken }) => { 
+      this.rows.push(...rows);                                                                                      
+      this.stats = stats;
+      this._changeDetectorRef.detectChanges();
```

Comment:
What's the difference between forcing detectChanges here, and the "force an update" code above?

Reply:
This triggers the ngDoCheck() function in the map component. That function does not update the map every time due to the batching behavior that it has.

The update above is to do the final update even if we are in the middle of the batch because the query is done.